app/models/user.rb

```diff
@@ -23,6 +23,7 @@ class User < ActiveRecord::Base
 
   before_create :set_token
   validates :time_format, inclusion: { in: TIME_FORMATS }
+  validates :external_id, presence: :true, unless: :integration?
```

Comment:
How about we validate on `create` only, so we don't end up with a situation where existing users become unable to save their user objects?